webapp/channels/src/components/plugin_link_tooltip/index.tsx

```diff
@@ -0,0 +1,135 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {
+    useFloating,
+    autoUpdate,
+    safePolygon,
+    useHover,
+    useDismiss,
+    useInteractions,
+    FloatingPortal,
+    autoPlacement,
+    inline,
+    useTransitionStyles,
+    FloatingOverlay,
+    FloatingFocusManager,
+    arrow,
+    offset,
+    FloatingArrow,
+    useFocus,
+    useRole,
+} from '@floating-ui/react';
+import classNames from 'classnames';
+import React, {useRef, useState} from 'react';
+import type {AnchorHTMLAttributes, ReactElement} from 'react';
+
+import Pluggable from 'plugins/pluggable';
+import {RootHtmlPortalId, OverlaysTimings, OverlayArrow, A11yClassNames} from 'utils/constants';
+
+import './plugin_link_tooltip.scss';
+
+interface Props {
+    nodeAttributes: AnchorHTMLAttributes<HTMLAnchorElement>;
+    children: ReactElement;
+}
+
+function PluginLinkTooltip(props: Props) {
+    const [isOpen, setOpen] = useState(false);
+
+    const arrowRef = useRef(null);
+
+    const {refs: {setReference, setFloating}, floatingStyles, context: floatingContext} = useFloating({
+        open: isOpen,
+        onOpenChange: setOpen,
+        whileElementsMounted: autoUpdate,
+        middleware: [
+            offset(OverlayArrow.OFFSET),
+            inline(),
+            autoPlacement({
+                allowedPlacements: ['top', 'bottom'],
+            }),
+            arrow({
+                element: arrowRef,
+            }),
+        ],
+    });
+
+    const {isMounted, styles: transitionStyles} = useTransitionStyles(floatingContext, {
+        duration: {
```

Comment:
0/5, but would it make sense to have this whole `{open: ..., close: ...}` object be the constant?